Sources/SystemInternals/Syscalls.swift

```diff
@@ -35,11 +38,19 @@ private func mockImpl(
   switch driver.forceErrno {
   case .none: break
   case .always(let e):
+#if os(Windows)
+    _ = ucrt._set_errno(e)
+#else
     errno = e
+#endif
     return -1
   case .counted(let e, let count):
     assert(count >= 1)
+#if os(Windows)
+    _ = ucrt._set_errno(e)
+#else
```

Comment:
I'm definitely in favor of making a helper function to set errno if it means we can avoid scattering `#if/#else` throughout function bodies. But that can be a subsequent PR.